### ISSUE TITLE
Set $container_uuid for systemd containers

### ISSUE THREAD

**Issue by jfroy** [containers/podman#13187]
/kind feature

**Description**

Set `$container_uuid` for containers running in systemd mode. From https://systemd.io/CONTAINER_INTERFACE/:

> systemd has special support for allowing container managers to initialize the UUID for /etc/machine-id to some manager supplied value. This is only enabled if /etc/machine-id is empty (i.e. not yet set) at boot time of the container. The container manager should set $container_uuid as environment variable for the container’s PID 1 to the container UUID.

The UUID could be derived from the container ID's first 128 bits, or by mixing the full container ID's 256 bits. Doing it that way does mean that the `$container_uuid` variable would be set fairly late in the container creation process, perhaps in `Runtime::setupContainer`.

**Comment by rhatdan** [comment]
I agree this is a good idea.  I think you would just truncate the container id to 64 chars. I am still not sure what functionality this would add.

**Comment by jfroy** [comment]
It's a small thing for sure. Having a machine-id tied to the lifetime of the container is semantically what you want. I am not sure to what extent the dbus / systemd machine-id is used by software, but on the flip side it exists and it should be as correct as can be.

**Comment by jfroy** [comment]
I'm happy to send a PR, but I wanted to open an issue first to discuss wisdom (do we want this) and strategy (I am not very familiar with podman or libpod).

**Comment by rhatdan** [comment]
Open a PR, I think it makes sense.

**Comment by jdoss** [comment]
May I add a suggestion to this feature request? 

>The /run/host/container-uuid file may be used to pass the same information as the $container_uuid environment variable (see above). This file should be newline terminated.

This would be super helpful too. Since ENV vars can be overwritten, an immutable file that has a container UUID would be awesome to have available inside the container.

**Comment by jfroy** [comment]
> May I add a suggestion to this feature request?
> 
> > The /run/host/container-uuid file may be used to pass the same information as the $container_uuid environment variable (see above). This file should be newline terminated.
> 
> This would be super helpful too. Since ENV vars can be overwritten, an immutable file that has a container UUID would be awesome to have available inside the container.

I considered adding this file as well, but looking at the systemd code, it is never read. I suppose it exists for other software. In any case, it seems like a good addition and I'll see if I can spend the time to add it.
